src/gmail.js

```diff
@@ -3843,19 +3878,36 @@ var Gmail = function(localJQuery) {
         */
         recipients: function(options) {
             if( typeof options !== "object" ) options = {};
-            var name_selector = options.type ? "[name=" + options.type + "]" : "";
 
-            // determine an array of all emails specified for To, CC and BCC and extract addresses into an object for the callback
-            var recipients = options.flat ? [] : { to: [], cc: [], bcc: [] };
-            this.$el.find(".GS input[type=hidden]"+name_selector).each(function(idx, recipient ){
-                if(options.flat) {
-                    recipients.push(recipient.value);
-                } else {
-                    if(!recipients[recipient.name]) recipients[recipient.name] = [];
-                    recipients[recipient.name].push(recipient.value);
-                }
-            });
-            return recipients;
+            // Peoplekit
+            if (this.$el.find("div[data-hovercard-id]").length) {
+                let type_selector = options.type ? " div[name=" + options.type + "]" : "";
+                let recipients = options.flat ? [] : { to: [], cc: [], bcc: [] };
+                this.$el.find("tr.bzf" + type_selector + " div[data-hovercard-id]").each((_, el) => {
+                    const email = el.getAttribute("data-hovercard-id");
+                    if (options.flat) {
+                        recipients.push(email);
+                    } else {
+                        const type = el.closest("div[name]").getAttribute("name");
+                        if(!recipients[type]) recipients[type] = [];
+                        recipients[type].push(email);
+                    }
+                });
+                return recipients;
+            } else {
+                var name_selector = options.type ? "[name=" + options.type + "]" : "";
+                // determine an array of all emails specified for To, CC and BCC and extract addresses into an object for the callback
+                var recipients = options.flat ? [] : { to: [], cc: [], bcc: [] };
+                this.$el.find(".GS input[type=hidden]"+name_selector).each(function(idx, recipient ){
+                    if(options.flat) {
+                        recipients.push(recipient.value);
+                    } else {
+                        if(!recipients[recipient.name]) recipients[recipient.name] = [];
+                        recipients[recipient.name].push(recipient.value);
+                    }
+                });
+                return recipients;
```

Comment:
There's quite a bit of duplication here, of what I will admit I find to be non-great code, and doubly admit it's definitely not your fault.

What bothers me most right now is probably the flat vs non-flat aspect, and that seems like something which could be split out quite easily:

````javascript
let recipients = [];

if (isPeopleKit()) {
    recpieints = this.$el.find("peopleKit-selector).map( i => i.getAttribute(""));
} else {
    recpieints = this.$el.find("nonpeopleKit-selector);
}

if (options.flat) {
    return recipients;
} else {
    let result = {};
    result[options.type] = recipients;
    return result;
}
````

I think that would be much cleaner. What do you think?